lib/routes/account.js

```diff
@@ -492,13 +520,36 @@ module.exports = function (
           if (shouldSendNewDeviceLoginEmail) {
             // The response doesn't have to wait for this,
             // so we don't return the promise.
-            mailer.sendNewDeviceLoginNotification(
-              emailRecord.email,
-              userAgent.call({
-                acceptLanguage: request.app.acceptLanguage,
-                timestamp: Date.now()
-              }, request.headers['user-agent'])
-            )
+            var location = null
+            var timeZone = null
+            getGeoData(ip, request)
+              .then(
+                function (loc) {
+                  location = {
+                    city: loc.city,
+                    country: loc.country
+                  }
+                  timeZone = loc.timeZone
+                })
+              .catch(
+                function (err) {
+                  // log error stuff
+                  log.error({ op: 'mailer.sendNewDeviceLoginNotification.1', err: err.message})
+                }
+              )
+              .then(
```

Comment:
@vladikoff Which global variables? Also, the wrapper might throw an error, which is the reason for the .catch instead of a `rejection handler`.